src/map/spell.cpp

```diff
@@ -696,8 +696,9 @@ namespace spell
                     return false;
                 }
 
-                // ensure trust level is appropriate+
-                if (PCaster->objtype == TYPE_TRUST && PCaster->GetMLevel() < static_cast<CMobEntity*>(PCaster)->m_SpellListContainer->GetSpellMinLevel(spell->getID()))
+                // Ensure pet or trust is level appropriate
+                if (((PCaster->objtype == TYPE_PET && static_cast<CPetEntity*>(PCaster)->getPetType() != PET_TYPE::AUTOMATON) || PCaster->objtype == TYPE_TRUST) &&
```

Comment:
I have a copy of `spell.cpp` with those changes added and working. While I was looking over the commit, I realized that I could handle the different `objtype` cases with a switch. I got a second `spell.cpp` working with the switch statement, the change is drastic so I wanted to clear it with you before making the commit.
```c++
// Check If user can cast spell
bool CanUseSpell(CBattleEntity* PCaster, CSpell* spell)
{
    if (spell == nullptr)
    {
        return false;
    }

    bool usable = false;
    uint8 requirements = spell->getRequirements();

    switch(PCaster->objtype)
    {
        case TYPE_MOB:
            // Unable to cast because caster is hidden or untargetable
            if (PCaster->IsNameHidden() || static_cast<CMobEntity*>(PCaster)->GetUntargetable())
            {
                return false;
            }
            // Mobs can cast any non-given char spell
            return true;

        case TYPE_PC:
            if (spell->getSpellGroup() == SPELLGROUP_TRUST)
            {
                return true; // every PC can use trusts
            }
            else if (luautils::OnCanUseSpell(PCaster, spell))
            {
                return true;
            }
            [[fallthrough]];
        case TYPE_FELLOW:
        case TYPE_NPC:
        case TYPE_SHIP: // This can be removed it does technically follow current logic though
            // Make sure caster has the right main job and level
            if (PCaster->GetMLevel() >= spell->getJob(PCaster->GetMJob()))
```
Removed what goes between here because it's the current code.
```c++
        case TYPE_PET:
        case TYPE_TRUST:
            // Unable to cast because caster is hidden or untargetable
            if (PCaster->IsNameHidden() || static_cast<CMobEntity*>(PCaster)->GetUntargetable())
            {
                return false;
            }

            if (static_cast<CPetEntity*>(PCaster)->getPetType() == PET_TYPE::AUTOMATON)
            {
                return true;
            }

            // Ensure pet or trust is level appropriate
            if (PCaster->GetMLevel() < static_cast<CMobEntity*>(PCaster)->m_SpellListContainer->GetSpellMinLevel(spell->getID()))
            {
                return false;
            }
            return true;
```